app/main.py

```diff
@@ -1,6 +1,17 @@
 from typing import Callable
 
+saved_runs = {}
+
```

Comment:
Need to move `saved_runs` inside `cache` function

Reply:
done